src/test/java/duke/ParserTest.java

```diff
@@ -0,0 +1,256 @@
+        assertTrue(Parser.parse("list") instanceof ListCommand);
+    }
+
+    @Test
+    public void parse_listInvalid_exceptionThrown() {
+        try {
+            Parser.parse("list bye");
+            fail();
+        } catch (DukeException e) {
+            assertEquals("I'm sorry, but I don't know what that means :-(", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parse_todo_success() throws DukeException {
+        assertTrue(Parser.parse("todo sleep") instanceof TodoCommand);
+    }
+
+    @Test
+    public void parse_todoNoInput_exceptionThrown() {
+        try {
+            Parser.parse("todo");
+            fail();
+        } catch (DukeException e) {
+            assertEquals("Please enter a task todo.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parse_event_success() throws DukeException {
+        assertTrue(Parser.parse("event sleep /at 2022-8-22 1000") instanceof EventCommand);
+    }
+
+    @Test
+    public void parse_eventNoTime_exceptionThrown() {
+        try {
+            Parser.parse("event sleep");
+            fail();
+        } catch (DukeException e) {
+            assertEquals("Please use /at to specify event time.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parse_eventNoInput_exceptionThrown() {
+        try {
+            Parser.parse("event");
+            fail();
+        } catch (DukeException e) {
+            assertEquals("Please enter an event.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parse_deadline_success() throws DukeException {
+        assertTrue(Parser.parse("event sleep /by 2022-8-22 1000") instanceof DeadlineCommand);
+    }
+
+    @Test
+    public void parse_deadlineNoTime_exceptionThrown() {
+        try {
+            Parser.parse("deadline fight");
+            fail();
+        } catch (DukeException e) {
+            assertEquals("Please use /by to specify event time.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parse_deadlineNoInput_exceptionThrown() {
+        try {
+            Parser.parse("deadline");
+            fail();
+        } catch (DukeException e) {
+            assertEquals("Please enter a deadline.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parse_date_success() throws DukeException {
+        assertTrue(Parser.parse("date 2022-08-22") instanceof DateCommand);
+    }
+
+    @Test
+    public void parse_dateInvalid_exceptionThrown() {
+        try {
+            Parser.parse("date 1");
+            fail();
+        } catch (DukeException e) {
+            assertEquals("Please enter date in yyyy-M-d format.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parse_dateNoInput_exceptionThrown() {
+        try {
+            Parser.parse("date");
+            fail();
+        } catch (DukeException e) {
+            assertEquals("Please enter a date.", e.getMessage());
+        }
+    }
+}
```

Comment:
I like that you wrote a comprehensive list of tests for your parse function, and they are all named in the right format!